game.py

```diff
@@ -89,12 +76,38 @@ def print_word_progress_string(snowman_word, snowman_word_dict):
         output_string += " "
     print(output_string)
 
-def get_word_progress(snowman_word, snowman_word_dict):
-    """
-    This function takes the snowman_word and snowman_word_dict as input.
-    It returns True if all the letters of the word have been guessed, and False otherwise.
-    """
-    for letter in snowman_word:
-        if not snowman_word_dict[letter]:
-            return False
-    return True
\ No newline at end of file
+
+def snowman(snowman_word):
+    r = RandomWord()
+    snowman_word = r.word(
+        word_min_length=SNOWMAN_MIN_WORD_LENGTH, 
+        word_max_length=SNOWMAN_MAX_WORD_LENGTH)
+
+    print(f"debug info: {snowman_word}")
+
+    correct_guesses_list = []
```

Comment:
👀 Through the Learn material, we end up replacing the idea of a correct list with a word dictionary, that tracks whether each letter has been guessed in an efficient way. A helper method, `build_word_dict` is supplied to help construct the dictionary. `get_letter_from_user`, `get_word_progress`, and `print_word_progress_string` all expect the dictionary built from `build_word_dict` to be passed as one of the arguments.